fixed-point.rkt

```diff
@@ -0,0 +1,11 @@
+#lang racket
+
+(define (fixed-point f next x precision maxiter)
```

Comment:
lisp is easier to parse (with our brain) with some conventions in s-expr's organization. 

A good practice is never put '(' and ')' lone in a line, because the entire expression needs to fit in our brain to parse lisp and adding new lines with lonely parens only makes it harder to find the pattern (is this a function? is this a macro? and so on).

Take a look here: http://community.schemewiki.org/cgi-bin/scheme.cgi?scheme-style

Another important aspect of reading lisp code is that our brain parses it by pattern-matching the whole expression and then having a concise structure in how you write the expression can help a lot. For example, a _if expression_ has commonly the pattern:

```racket
(if <cond>
    <true expression>
    <false expression>)
```

Then, as this is the accepted way (by style) of writing an if expression, should be much more easy to read:

```racket
(if (>= a b)
    a
    b)
```
than this:
```racket
(if (>= a b) a
    b)
```
Both are easily recognizable in reading alone, but hard to recognize the latter when surrounded by other kinds of expressions and/or macros (the good thing is that our brain detects the anomaly also, and force us to read more carefully the strange expression).

Then, an alternative way (in regard to scheme style) to write this function could be:

```racket
(define (fixed-point f next x precision maxiter)
    (if (<= maxiter 0)
        x
        (if (<= (f x) precision)
            x
            (fixed-point f next (next x) precision (- maxiter 1)))))
```
And maybe another way, to avoid the repetition of returns, is using _and_ and _cond_:

```racket
(define (fixed-point f next x precision maxiter)
    (cond 
      [(and (> maxiter 0)
            (> (f x) precision)
            (fixed-point f next (next x) precision (- maxiter 1)))]
      [else x]))
```

The _cond_ could be avoided also using _or_.
```racket
(or (and ...) x)
```

But I think the _cond_ is much more expressive in this case.

But feel free to ignore all of this, as is everything about style and you should use what you are most comfortable with. But I think you could have some trouble reading other lispers code because your brain will not be tuned for them.

Reply:
Thanks, it actually makes sense. It also makes me understand better why the Guido developed python with the identation as part of the syntax, lisp relies heavily on using this as a style to be more readable, it was something that always pleased me on Python when I started developing on it. These days I understand the tradeoffs better...but even so...python code can be pretty good to read because of this enforcement. Thanks bro =D